### PR TITLE
docs: fix reference to account role resource

### PR DESCRIPTION
**Pull request by joscha**

### CODE REVIEW

File: docs/resources/grant_account_role.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ resource "snowflake_user" "user" {
 }
 
 resource "snowflake_grant_account_role" "g" {
-  role_name = snowflake_role.role.name
+  role_name = snowflake_account_role.role.name
```
Comment (sfc-gh-jcieslak) [inline docs/resources/grant_account_role.md:46]: Hey @joscha 
Please make the same change in examples/resources/snowflake_grant_account_role/resource.tf. That would be sufficient to merge the pr.
If you would like to make future contributions to the documentation, please mind that it's generated. In this case, you could only modify examples/resources/snowflake_grant_account_role/resource.tf and run `make docs` to generate the .md file you changed manually here. Before running `make docs` it's necessary to run `make dev-setup` before, to download the tool responsible for doc generation.

**Review by sfc-gh-jcieslak — CHANGES_REQUESTED** [review]

File: docs/resources/grant_account_role.md (hunk 1 above)
Comment (joscha) [reply docs/resources/grant_account_role.md:46]: I see, thank you, done in [19e64d6](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3101/commits/19e64d64fa8b3b842452b85602962b60e8658043)

**Comment by sfc-gh-jmichalak** [comment]
Thanks for your contribution! LGTM, but the PR title is not valid. Please use `docs(examples):` or simply `docs:` as a prefix.

**Comment by joscha** [comment]
> Thanks for your contribution! LGTM, but the PR title is not valid. Please use `docs(examples):` or simply `docs:` as a prefix.

I changed it and the ❌ disappeared on the build.

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Comment by joscha** [comment]
> Integration tests failure for [19e64d64fa8b3b842452b85602962b60e8658043](https://github.com/Snowflake-Labs/terraform-provider-snowflake/actions/runs/11009495394)

looks unrelated?
```
2024/09/24 08:17:26 Integration test context initialisation failed with: `002002 (42710): SQL compilation error:
Object '"int_test_db_IT_19E64D64FA8B3B842452B85602962B60E8658043"' already exists.`
2024/09/24 08:17:26 Running integration tests cleanup
2024/09/24 08:17:26 [DEBUG] sql-conn-exec: [query DROP DATABASE IF EXISTS "int_test_db_IT_19E64D64FA8B3B842452B85602962B60E8658043" err <nil> duration 315.752794ms args {}] (YXA30390)
program not built with -cover
FAIL	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/testint	5.807s
```

**Comment by sfc-gh-jmichalak** [comment]
Yes, don't worry :)

**Pull request merged**